main.nf

```diff
@@ -635,111 +641,245 @@ process VCF_CONSENSUS {
+        if (params.variant == 'other_variant_calling _tool') {
 
-    if (params.align == 'bowtie2') {
-        
-        BOWTIE2(ch_ref, FASTP.out.reads)
-        BAMINDEX(BOWTIE2.out.align)
+            FREEBAYES(ref, BAMINDEX.out.align)
     
-    } else if (params.align == 'minimap2') {
-        
-        MINIMAP2(ch_ref, FASTP.out.reads)
-        BAMINDEX(MINIMAP2.out.align)
+        } else {
     
-    } else {
-        
-        BWA(ch_ref, FASTP.out.reads)
-        BAMINDEX(BWA.out.align)
-    }
+            FREEBAYES(ref, BAMINDEX.out.align)
     
-    //----------------------------------------
-    // VARIANT CALLING 
-    //----------------------------------------
+        }
     
-    if (params.variant == 'other_variant_calling _tool') {
-
-        FREEBAYES(ch_ref, BAMINDEX.out.align)
+        BCFTOOLS_STATS(FREEBAYES.out.variant)
     
-    } else {
+        if (params.filter == 'other_filtering_tool') {
+
+            BCFTOOLS_FILTER(FREEBAYES.out.variant)
     
-        FREEBAYES(ch_ref, BAMINDEX.out.align)
+        } else {
+        
+            BCFTOOLS_FILTER(FREEBAYES.out.variant)
+        }
+            
+    emit:
+        variant = BCFTOOLS_FILTER.out.variant
+
+}
+//----------------------------------------
+
+//----------------------------------------
+// WORKFLOW: consensus
+// Takes:
+//      variants = variant calls on reads
+//      reads = paired end reads
+//      ref = reference genome
+//      
+// Main: 
+//      1. Annotation && effect prediction
+//      2. Consensus
+//
+//----------------------------------------
+workflow consensus {
+
+    take:
+        variants
+        reads
+        ref
+
+    main: 
+       
+        if (params.prediction == 'snpeff') {
+            //deprecated until further notice        
+            SNPEFF(variants, reads)
+        } else {
+
+            SNIPPY(variants, reads)
+        }
     
-    }
+        if (params.consensus == 'bcftools') {
+
+            BCFTOOLS_CONSENSUS(ref, SNIPPY.out.annotation)
     
-    //----------------------------------------
-    // STATS
-    //----------------------------------------
+        } else {
 
-    BCFTOOLS_STATS(FREEBAYES.out.variant)
+            VCF_CONSENSUS(ref, SNIPPY.out.annotation)
     
-    //----------------------------------------
-    // FILTERING 
-    //----------------------------------------
+        }
+}
+//----------------------------------------
 
-    if (params.filter == 'other_filtering_tool') {
+//----------------------------------------
+// WORKFLOW: bwa
+// Takes:
+//      ref = reference genome
+//      reads = paired end reads
+//      phix = phix genome
+//      
+// Main: 
+//      1. Quality check and read filtering 
+//      2. Read mapping using BWA
+//      3. Variant calling
+//      4. Consensus 
+//
+//----------------------------------------
+workflow bwa {
+
+    take: 
+        ref
+        reads
+        phix
+
+    main:
+        quality_check(reads, phix)
```

Comment:
Could you refactor out the `quality_check` and have this workflow and the others like it accept quality trimmed/filtered reads?

Reply:
I have taken quality_check out of the bwa, bowtie2, and minimap2 workflows but left it in the main.nf workflow since the tuple the aligner processes expects is sample_id, r1, r2 and leaving quality_check ensures this and makes development easier. It will be simple to remove once dev is done